Example/PaymentSheet Example/PaymentSheet Example/EmbeddedPlaygroundViewController.swift

```diff
@@ -146,9 +155,34 @@ class EmbeddedPlaygroundViewController: UIViewController {
         ])
     }
 
-    func setSettingsView(_ settingsView: UIView) {
-        settingsViewContainer.arrangedSubviews.forEach { settingsViewContainer.removeArrangedSubview($0) }
-        settingsViewContainer.addArrangedSubview(settingsView)
+    func setSettingsView<SettingsView: View>(_ settingsView: @escaping () -> SettingsView) {
+        // Remove existing hosting controller if any
+        hostingController?.willMove(toParent: nil)
+        hostingController?.view.removeFromSuperview()
+        hostingController?.removeFromParent()
+
+        // Create new hosting controller
+        let rootView = settingsView().environmentObject(playgroundController)
+        hostingController = UIHostingController(rootView: AnyView(rootView))
+
+        guard let hostingController = hostingController else { return }
+
+        // Add as child view controller
+        addChild(hostingController)
+        settingsViewContainer.addArrangedSubview(hostingController.view)
+        hostingController.didMove(toParent: self)
+    }
+    
+    private func observePlaygroundController() {
+        playgroundController.objectWillChange
+            .sink { [weak self] _ in
+                DispatchQueue.main.async {
+                    self?.hostingController?.rootView = AnyView(
+                        EmbeddedSettingsView().environmentObject(self?.playgroundController ?? PlaygroundController(settings: PaymentSheetTestPlaygroundSettings.defaultValues()))
+                    )
```

Comment:
If we pass in a new instance of PlaygroundController, then seems like it would no longer be observed?

Reply:
Yep good point